### ISSUE TITLE
`four-forward-slashes` thinks module-level docs are items that can be documented

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#16168]
Using the following flags

`--force-warn clippy::four-forward-slashes`

this code:

```rust
pub fn main() {
    //// I am not a doc comment!
    //! another comment
}

```

caused the following diagnostics:

```
    Checking _a v0.1.0 (/tmp/icemaker_global_tempdir.ZJcEP8tl8ptu/icemaker_clippyfix_tempdir.U1dWplOlXAU9/_a)
warning: this item has comments with 4 forward slashes (`////`). These look like doc comments, but they aren't
 --> src/main.rs:1:1
  |
1 | / pub fn main() {
2 | |     //// I am not a doc comment!
3 | |     //! another comment
  | |_______________________^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#four_forward_slashes
  = note: requested on the command line with `--force-warn clippy::four-forward-slashes`
help: make this a doc comment by removing one `/`
  |
2 -     //// I am not a doc comment!
3 -     //! another comment
2 + /// I am not a doc comment!
  |

warning: `_a` (bin "_a") generated 1 warning (run `cargo clippy --fix --bin "_a" -p _a` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.11s

```

However after applying these diagnostics, the resulting code:

```rust
pub fn main() {
/// I am not a doc comment!
    // another comment
}

```
no longer compiled:

```
    Checking _a v0.1.0 (/tmp/icemaker_global_tempdir.ZJcEP8tl8ptu/icemaker_clippyfix_tempdir.U1dWplOlXAU9/_a)
error[E0585]: found a documentation comment that doesn't document anything
 --> src/main.rs:2:1
  |
2 | /// I am not a doc comment!
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: doc comments must come before what they document, if a comment was intended use `//`

For more information about this error, try `rustc --explain E0585`.
error: could not compile `_a` (bin "_a") due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `_a` (bin "_a" test) due to 1 previous error

```

Version:
```
rustc 1.93.0-nightly (9b82a4fff 2025-12-01)
binary: rustc
commit-hash: 9b82a4fffe2b215f488a6dfdc0b508235f37c85c
commit-date: 2025-12-01
host: x86_64-unknown-linux-gnu
release: 1.93.0-nightly
LLVM version: 21.1.5

```
